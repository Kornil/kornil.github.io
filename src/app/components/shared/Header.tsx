import React from "react";

import { NavLink, StyledHeader } from "app/styled-components";

const Header: React.SFC<{}> = () => (
  <StyledHeader>
    <NavLink activeClassName="active" exact to="/">
      home
    </NavLink>
    <NavLink activeClassName="active" to="/blog">
      blog
    </NavLink>
    <NavLink activeClassName="active" to="/contact">
      contact
    </NavLink>
  </StyledHeader>
);

export default Header;
