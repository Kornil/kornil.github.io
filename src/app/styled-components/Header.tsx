import styled from "react-emotion";

const Header = styled.header`
  grid-area: nav;
  padding: 8px 0;
`;

export default Header;
