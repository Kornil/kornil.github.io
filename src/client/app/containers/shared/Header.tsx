import React from "react";
import styled from "react-emotion";

import { NavLink } from "app/styled-components";

const StyledHeader = styled.header`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  padding: 8px 0;
  height: 40px;
  align-items: center;

  a:nth-of-type(1) {
    text-align: end;
  }
  a:nth-of-type(3) {
    text-align: start;
  }
`;

const Header: React.SFC<{}> = () => (
  <StyledHeader>
    <NavLink activeClassName="active" exact to="/">
      home
    </NavLink>
    <NavLink activeClassName="active" to="/blog">
      blog
    </NavLink>
    <NavLink activeClassName="active" to="/contact">
      contact
    </NavLink>
  </StyledHeader>
);

export default Header;
