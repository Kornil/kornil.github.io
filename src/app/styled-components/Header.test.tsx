import { shallow } from "enzyme";
import React from "react";

import Header from "./Header";

test("it works", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
