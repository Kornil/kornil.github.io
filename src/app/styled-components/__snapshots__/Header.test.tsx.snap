// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  grid-area: nav;
  padding: 8px 0;
}

<header
  className="emotion-0 emotion-1"
/>
`;
